scripts/xdist/setup_worker.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Set up worker node.
+
+while getopts 'p:d:' opt; do
+    case "$opt" in
+	p) PYTHON_VERSION="$OPTARG";;
+	d) DJANGO_REQUIREMENT="$OPTARG";;
+	[?])
+	    print >&2 "Usage: $0 -p python-version -d django-reqs-file"
+	    exit 1
+            ;;
+    esac
+done
+
+venv_parent=/home/jenkins/edx-venv-${PYTHON_VERSION}
+venv=$venv_parent/edx-venv
+rm -rf $venv
+tar -C $venv_parent -xf /home/jenkins/edx-venv_clean-${PYTHON_VERSION}.tar.gz
+source $venv/bin/activate
+
+# Hack to fix up egg-link files given that the virtualenv is not relocatable
+sed -i "s|\(^/home/jenkins\)/shallow-clone|\1/edx-platform|" -- \
+    $venv/lib/python*/site-packages/*.egg-link
+pip install -qr requirements/edx/pip-tools.txt
+pip-sync requirements/edx/testing.txt "${DJANGO_REQUIREMENT}"
```

Comment:
@jmbowman I feel like there was a reason why we don't do this but also tests seem to be passing.  Do you remember the reason?

Reply:
I think this would be ideal for fixing our current issue, if there are no downsides.  Thanks @timmc-edx.

Reply:
@jinder1s was the last person to try this in edx-platform: https://github.com/edx/edx-platform/pull/22315 .  Maybe the egg-link hack fixes the issue he encountered?